Add Floccus Bookmarks to Markdown

mddevils opened the pull request:
Add floccus-bookmarks-to-markdown

# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

ericaxu commented:
Hi @mddevils , thanks for submitting your plugin!

Here are some suggestions:

[addClass('my-plugin-ribbon-class')](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L43)
Use a more descriptive and specific class name

[const xbelFilePath: string = join(xbelFolderPath, xbelFileName);](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L80)
Consider writing a util function yourself for simple things like join to make it work on mobile. Or is it intentional that it doesn't work on mobile for the use case of this plugin?

[this.app.vault.createFolder(mdFolderPath);](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L88)
Add `await`

[backupExistingFile](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L116)
async function

[createFolder](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L126)
[rename](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L136)
[trash](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L161)

`await` in all these places to make sure everything happens one after another. For anything in our API that returns a Promise, use await to make sure it's finished before the next line executes.

[containerEl.createEl('h2', { text: 'Floccus Bookmarks to Markdown Settings' });](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L216)
This is not necessary

Everything else looks okay. Double checking all the async functions and properly await them should resolve most of the concerns.

mddevils commented:
@ericaxu, Thank you so much for your time and for the feedback! 

I made new release 0.2.1 with the suggestions provided:
- [x] [addClass('my-plugin-ribbon-class')](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L43) Use a more descriptive and specific class name
'my-plugin-ribbon-class' to 'floccus-bookmarks-to-md-icon'

- [x] [this.app.vault.createFolder(mdFolderPath);](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L88) Add `await`

- [x] [createFolder](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L126) [rename](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L136) [trash](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L161) `await` in all these places to make sure everything happens one after another. For anything in our API that returns a Promise, use await to make sure it's finished before the next line executes.
Added `await` to 'createFolder', 'rename' and 'trash' calls

- [x] [backupExistingFile](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L116) async function
Added `async` to 'backupExistingFile'

- [x] [containerEl.createEl('h2', { text: 'Floccus Bookmarks to Markdown Settings' });](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L216) This is not necessary
containerEl.createEl('h2', { text: 'Floccus Bookmarks to Markdown Settings' });

- [ ] [const xbelFilePath: string = join(xbelFolderPath, xbelFileName);](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/0db90a0c50930ad2c2c11418382cd5ec537ec4a6/main.ts#L80) Consider writing a util function yourself for simple things like join to make it work on mobile. Or is it intentional that it doesn't work on mobile for the use case of this plugin?

I may consider make it available for  mobile in the future.

ericaxu commented:
Looks good! Thanks for making the changes so quickly. LGTM, passing along to @joethei .

mddevils commented:
@joethei , Thank you so much for your time and for the feedback!

I made new release 0.2.2 with the suggestion provided:
[.setName('XBEL Absolute Folder Path')](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/850f6170a63def3bc33a10d330461ffdf4f9dfda/main.ts#L217)
- [x] [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

joethei commented:
[const mdFolder = this.app.vault.getAbstractFileByPath(mdFolderPath) as TFolder;](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/63c490b1a2c681742d3d4dc5e8c3effcc816c7c1/main.ts#L85-L88)
Don't blindly cast, instead do this:
```ts
const mdFolder = this.app.vault.getAbstractFileByPath(mdFolderPath);
if (!mdFolder instanceof TFolder) {
	await this.app.vault.createFolder(mdFolderPath);
 }
```

liamcain commented:
- [updateInterval](https://github.com/mddevils/floccus-bookmarks-to-markdown/blob/63c490b1a2c681742d3d4dc5e8c3effcc816c7c1/main.ts#L49) add a check to make sure that updateInterval > 0
